### PR TITLE
Difference of squares:   sync expected test results and input data with problem-specifications

### PR DESCRIPTION
**Pull request by BethanyG**
Part of https://github.com/exercism/python/issues/1762

- Changed function name to  **difference_of_squares** in test cases, exercise stub, and `example.py` to conform with `canonical-data.json`.
- Changed `example.py` to use the formulas from https://brilliant.org/wiki/sum-of-n-n2-or-n3/, rather than loops.
- Changed argument name in `example.py` and exercise stub to **number** from _count_.

### CODE REVIEW

File: exercises/difference-of-squares/difference_of_squares_test.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import unittest
 
-from difference_of_squares import difference, square_of_sum, sum_of_squares
+from difference_of_squares import difference_of_squares, \
+     square_of_sum, sum_of_squares
```
[the remark this answers is not in the document]
Comment (BethanyG) [reply exercises/difference-of-squares/difference_of_squares_test.py]: Ooops.  I did not see this, or I wouldn't have closed this and deleted the branch.  I am an idiot.  :P  Will re-open and correct!

File: exercises/difference-of-squares/example.py [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+def square_of_sum(number):
+    sum_ = number * (number + 1) / 2
     return sum_ * sum_
 
 
-def sum_of_squares(count):
-    return sum(m * m for m in range(count + 1))
+def sum_of_squares(number):
+    numerator = number * (number + 1) * (2 * number + 1)
+    return numerator/6
```
Comment (cmccandless) [inline exercises/difference-of-squares/example.py]: Your `flake8` setup didn't complain about this? That is strange.

```suggestion
    return numerator / 6
```

Comment (BethanyG) [reply exercises/difference-of-squares/example.py]: I just ran `Flake8` again on the branch....**it did not complain!**.   Neither did PyCharm.  Is there a config file you can share for `Flake8`?  Currently I am using all defaults and setting `max-complextiy` to 10.  PyCharms `PEP8` enforcement appears to be turned on, but it only complains if the space around an operator is **asymmetric** - it doesn't care if the space is _missing_, only that it _matches_.  Grr.  _In any case_  noted for next time!

**Comment by cmccandless** [comment]
Merged; thanks for working on this!

**Pull request merged**